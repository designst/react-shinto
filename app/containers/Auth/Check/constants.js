import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
