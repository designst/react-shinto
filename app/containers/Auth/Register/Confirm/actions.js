import createAction from 'utils/createAction';

import {
  AUTH_REGISTER_CONFIRM_REQUEST,
  AUTH_REGISTER_CONFIRM_SUCCESS,
  AUTH_REGISTER_CONFIRM_FAILURE,
} from './constants';

export const registerConfirmRequest = createAction(AUTH_REGISTER_CONFIRM_REQUEST);
export const registerConfirmSuccess = createAction(AUTH_REGISTER_CONFIRM_SUCCESS);
export const registerConfirmFailure = createAction(AUTH_REGISTER_CONFIRM_FAILURE);

export const registerConfirmRequestWait = createAction(
  AUTH_REGISTER_CONFIRM_REQUEST,
  AUTH_REGISTER_CONFIRM_SUCCESS,
  AUTH_REGISTER_CONFIRM_FAILURE,
);
