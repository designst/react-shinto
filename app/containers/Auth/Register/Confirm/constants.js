import createConstant from 'utils/createConstant';

export const AUTH_REGISTER_CONFIRM_REQUEST = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_REQUEST',
);
export const AUTH_REGISTER_CONFIRM_SUCCESS = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_SUCCESS',
);
export const AUTH_REGISTER_CONFIRM_FAILURE = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_FAILURE',
);
