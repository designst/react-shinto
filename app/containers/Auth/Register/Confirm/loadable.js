/* @flow */

import loadable from 'loadable-components';

import Loading from 'components/Loading';
import ErrorDisplay from 'components/ErrorDisplay';

export default loadable(() => import(/* webpackChunkName: 'auth/register/confirm' */ './index'), {
  LoadingComponent: Loading,
  ErrorComponent: ErrorDisplay,
});
