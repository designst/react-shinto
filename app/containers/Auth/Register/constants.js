import createConstant from 'utils/createConstant';

export const AUTH_REGISTER_REQUEST = createConstant(__filename, 'AUTH_REGISTER_REQUEST');
export const AUTH_REGISTER_SUCCESS = createConstant(__filename, 'AUTH_REGISTER_SUCCESS');
export const AUTH_REGISTER_FAILURE = createConstant(__filename, 'AUTH_REGISTER_FAILURE');
