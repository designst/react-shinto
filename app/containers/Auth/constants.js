import createConstant from 'utils/createConstant';

export const AUTH_API_REQUEST = createConstant(__filename, 'AUTH_API_REQUEST');
export const AUTH_API_SUCCESS = createConstant(__filename, 'AUTH_API_SUCCESS');
export const AUTH_API_FAILURE = createConstant(__filename, 'AUTH_API_FAILURE');
