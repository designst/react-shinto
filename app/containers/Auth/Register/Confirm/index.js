import { compose } from 'redux';
import { connect } from 'react-redux';

import RegisterConfirm from 'components/Auth/Register/Confirm';

import { registerConfirmRequest } from './actions';

const mapStateToProps = (state, { match }) => ({
  token: match.params.token,
});

const mapDispatchToProps = {
  onSubmit: registerConfirmRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(RegisterConfirm);
