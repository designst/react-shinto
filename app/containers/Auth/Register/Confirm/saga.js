import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_REGISTER_CONFIRM_REQUEST } from 'containers/Auth/Register/Confirm/constants';

import {
  registerConfirmSuccess,
  registerConfirmFailure,
} from 'containers/Auth/Register/Confirm/actions';

const logger = createLogger(__filename);

const authRegisterConfirmUrl = process.env.SHINTO_AUTH_REGISTER_CONFIRM_API_ENDPOINT;

export function* authRegisterConfirm(action) {
  const { token } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Register Confirm: %s', token);

  try {
    const response = yield call(apiService.post, authRegisterConfirmUrl, {
      token,
    });
    yield put(registerConfirmSuccess(response.data));
  } catch (err) {
    yield put(registerConfirmFailure(err));
  }
}

export default function* authRegisterConfirmData() {
  yield takeLatest(AUTH_REGISTER_CONFIRM_REQUEST, authRegisterConfirm);
}
