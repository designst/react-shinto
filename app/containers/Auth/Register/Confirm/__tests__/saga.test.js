import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import authRegisterConfirmData from 'containers/Auth/Register/Confirm/saga';
import { AUTH_REGISTER_CONFIRM_REQUEST } from 'containers/Auth/Register/Confirm/constants';

describe('authRegisterConfirmData', () => {
  let authRegisterConfirmGenerator;

  beforeEach(() => {
    authRegisterConfirmGenerator = authRegisterConfirmData();

    const selectDescriptor = authRegisterConfirmGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authRegisterConfirmSuccess action if it requests the data successfully', () => {
    const token = 'confirm';

    return expectSaga(authRegisterConfirmData)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/register/confirm'
              ? {
                  data: {
                    token,
                  },
                }
              : {},
        }),
      })
      .dispatch({
        type: AUTH_REGISTER_CONFIRM_REQUEST,
        payload: {
          token,
        },
      })
      .withReducer(authReducer)
      .run();
  });

  it('should dispatch the authRegisterConfirmFailure action if it requests the data unsuccessfully', () => {
    const token = 'confirm';

    return expectSaga(authRegisterConfirmData)
      .dispatch({
        type: AUTH_REGISTER_CONFIRM_REQUEST,
        payload: {
          token,
        },
      })
      .withReducer(authReducer)
      .run();
  });
});
