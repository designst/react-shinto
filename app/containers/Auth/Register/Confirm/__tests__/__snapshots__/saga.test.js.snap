// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authRegisterConfirmData should dispatch the authRegisterConfirmFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/register/confirm/constants/AUTH_REGISTER_CONFIRM_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`authRegisterConfirmData should dispatch the authRegisterConfirmSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/register/confirm/constants/AUTH_REGISTER_CONFIRM_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
